### DISCUSSION TITLE
How track Hubspot form submit?

### DISCUSSION THREAD

**Discussion by JeremC7** [plausible/analytics#2439] [Anything Else]
Here an example of the Hubspot HTML code:
<script charset="utf-8" type="text/javascript" src="//js-eu1.hsforms.net/forms/embed/v2.js"></script>
<script>
  hbspt.forms.create({
    region: "eu1",
    portalId: "999999999",
    formId: "9999999-ffef-4f91-af01-999999"
  });
</script>

**Comment by RobertJoonas** [comment]
Thanks for reaching out @JeremC7!

We are currently working on a new script extension called `tagged-events` which will allow you to track forms by giving them plausible-specific class names. The feature is mostly ready but it's still waiting for some reviews from our team at the moment. Here's the GitHub pull request: https://github.com/plausible/analytics/pull/2333. And a pending documentation update with instructions how to use it: https://github.com/plausible/docs/pull/295.

It seems like the form you want to track is created dynamically with this snippet, so adding class names to the form will not be possible. However, assuming that the `formId: "9999999-ffef-4f91-af01-999999"` becomes the `id` attribute of the form, you should be able to add the class names dynamically with JavaScript as well (this process is explained in the instructions I linked to).

We're hoping to pre-release this feature soon, hopefully by the end of the month. We would appreciate it if you could test our solution on your site, tracking your form, and let us know your feedback. Thanks! :)

**Reply by JeremC7** [reply]
Thank you @RobertJoonas. I'm not sure where in the doc I can refer to in order to try adding the class names dynamically with JavaScript using the id attribute of the form. My technical skills are limited. 

Otherwise, I can wait for the release of the 'tagged-events' script extension.
